resources/player_override.ts

```diff
@@ -15,10 +15,10 @@ import Util from './util';
 // jobs remotely due to gauge data being local and many bits of information
 // loaded from memory.
 
-type PlayerChangedDetail = { detail: PlayerChangedRet };
+export type PlayerChangedDetail = { detail: PlayerChangedRet };
 type PlayerChangedFunc = (e: PlayerChangedDetail) => void;
 
-export const addPlayerChangedOverrideListener = (playerName: string,
+export const addPlayerChangedOverrideListener = (playerName: string | null,
```

Comment:
This isn't an actionable thing, just a thought I've had a few times, but I wonder if all the places we're doing `| null` could use a re-examination.

Reply:
I thought about adjusting this to be optional instead, but it can't be specified as `playerName?: string` because the second parameter is required still?

Reply:
Definitely not in this PR, but we could swap the order of the parameters? Also, this isn't targeted at this line of code, I'm just curious how much `null` should come up (outside of WebComponents, which is beyond our control). In my day to day work, I don't feel like I see `null` used often.

Reply:
I can add a TODO for that change and someone can follow up in a later PR, after the popup-text stuff is merged?

Reply:
I don't think it's necessary to add a TODO, more so I saw this and it popped back in my head that I'm curious is `| null` is an anti-pattern or if it's fine overall. Again, not really an actionable item for this PR, just thoughts.
@quisquous if they have thoughts.

Reply:
Unresolving this so that quis can weigh in and we can continue the discussion. I think that it makes more sense to make playerName an optional second parameter, myself. I just don't think it fits for this PR, since it would have to be updated downstream as well.

Reply:
Agreed with the general idea here.  I think my feelings on null and undefined are:
* `null | undefined` is a code smell, because you shouldn't need two "unset" values
* prefer `undefined` over `null` where possible, because it feels more "ergonomic" in a TypeScript sense (e.g. `?` for optionally undefined things, but no similar thing for `null`)
* if you have to use `null` (e.g. HTML stuff), then use `null` instead of `undefined`

I think a TODO is not useful in this case because there's a lot of places where we do this, and so this one TODO on this one spot feels a bit out of place.  If you'd like to make sure we don't lose this, you could file a "take a look at null vs undefined and clean things up" issue.

Reply:
I think the TODO is okay here because this is only used for popup-text?
I can just roll that fix into the popup-text conversion as well?

Reply:
Ah sure, sorry I misunderstood what you meant.  Sounds good.